### ISSUE TITLE
Unable to use `Widget()` inside Shiny

### ISSUE THREAD

**Issue by cpsievert** [manzt/quak#30]
I just took a quick look into rendering `quak.Widget()` in Shiny, and I get the following JS error:

```python
import polars as pl
import shiny.express
from shinywidgets import render_widget
import quak

df = pl.read_parquet("https://github.com/uwdata/mosaic/raw/main/data/athletes.parquet")

@render_widget
def _():
    return quak.Widget(df)
```

<img width="937" alt="Screenshot 2024-07-30 at 4 20 15 PM" src="https://github.com/user-attachments/assets/98b5261e-6b5d-48fd-985c-07395f170bc6">

What's interesting here is that [`./lib/widget.ts`'s import](https://github.com/manzt/quak/blob/a86eeefd396d5bfc4a7789671f84908dd9e8fc29/lib/widget.ts#L5) of `apache-arrow` appears to for stuff under `ipc/*`, but not for [this line of `io/stream.ts`](https://github.com/apache/arrow/blob/7e50097ba4239cf9368b77f438d877d4176141c9/js/src/io/stream.ts#L75). Any idea why that might be?

**Comment by manzt** [comment]
Sorry not an avid shiny user... can you provide how to run this code?

**Comment by cpsievert** [comment]
Oh interesting, I also get a similar error when running a [hello mosaic-widget](https://idl.uw.edu/mosaic/examples/weather.html) example:

```python
import pandas as pd
from mosaic_widget import MosaicWidget
import shiny.express
from shinywidgets import render_widget
import yaml

weather = pd.read_csv(
    "https://uwdata.github.io/mosaic-datasets/data/seattle-weather.csv",
    parse_dates=["date"],
)

# Load weather spec, remove data key to ensure load from Pandas
with open("weather.yaml") as f:
    spec = yaml.safe_load(f)
    _ = spec.pop("data")

@render_widget
def _():
    return MosaicWidget(spec, data={"weather": weather})

```

**Comment by manzt** [comment]
This sniffs of some runtime global inspection by arrow (or one of this dependencies). I'm guessing Shiny includes a dep that sets a global (e.g., `define` from requirejs) which changes the behavior between the front end environments.

**Comment by cpsievert** [comment]
Ohh, yea, that seems likely. `shiny`/`shinywidgets` puts `define` and `require` on the `window` so that it can do a run-time request path configuration (e.g., `window.require.config({"paths": {"anywidget": "lib/anywidget-0.0/index"}})`)

**Comment by manzt** [comment]
Just speculation. Don't know if that's it alone. I know that `require` and `define` are also in class Jupyter notebooks.

**Comment by manzt** [comment]
Ah, wait.... I'm being silly. I meant to bring this up on Friday. Shiny doesn't support sending binary data to/from the front end. This is essential for applications that use apache arrow.

**Comment by manzt** [comment]
I'm going to close. Feel free to re-open if there is an issue when/if shinywidgets adds support for binary data!

**Comment by cpsievert** [comment]
Ahh, nice catch, thank you!

**Comment by kylebarron** [comment]
Hmm I've gotten lonboard to work in shiny widgets I think

**Comment by manzt** [comment]
@kylebarron You are right (it base64 encodes the binary stuff). However, I’m guessing your handling of buffers is more flexible than quaks.

see https://github.com/posit-dev/py-shinywidgets/pull/152
